department-of-veterans-affairs/caseflow
# [PART 1] Remove attorney task cancellation hook

# FULL STACK

PART 1: Remove attorney task cancellation hook #14448
PART 2: Perform "caseflow" reassign of attorney tasks #14452
PART 3: Remove overtime status of appeal if case is reassigned #14453

# THIS PR

Resolves #14181

PART 1 in stack to implement #14366
PART 2 #14452
PART 3 #14453

### Description
Calls send_back_to_judge_assign! directly when cancelling an attorney task rather than relying on a callback.

### Acceptance Criteria
- [x] Hook for sending a case back to judge reassign on attorney task cancellation is removed.

### Testing Plan
1. Log in as BVACASPER
1. Go to any ama attorney task
1. Select "Cancel Task & return to judge"
1. Ensure the attorney and judge decision review task are cancelled and a judge assign task was opened
```ruby
uuid = ""
Appeal.find_by_uuid(uuid).treee
                                  ┌────────────────────────────────────────────────────────────────────────┐
Appeal 682 (evidence_submission)  │ ID   │ STATUS    │ ASGN_BY     │ ASGN_TO     │ UPDATED_AT              │
└── RootTask                      │ 2027 │ on_hold   │             │ Bva         │ 2020-05-22 14:50:19 UTC │
    ├── JudgeDecisionReviewTask   │ 2028 │ cancelled │             │ BVAAABSHIRE │ 2020-06-02 18:19:56 UTC │
    │   └── AttorneyTask          │ 2029 │ cancelled │ BVAAABSHIRE │ BVAEERDMAN  │ 2020-06-02 18:19:55 UTC │
    └── JudgeAssignTask           │ 2532 │ assigned  │             │ BVAAABSHIRE │ 2020-06-02 18:19:56 UTC │
                                  └────────────────────────────────────────────────────────────────────────┘
```
1. Cancel any attorney task by hand
```ruby
task = AttorneyTask.open.last
task.appeal.treee
Appeal 545 (direct_review) ──── │ ID   │ STATUS    │ ASGN_BY     │ ASGN_TO     │ UPDATED_AT              │
└── RootTask                    │ 1458 │ on_hold   │             │ Bva         │ 2020-05-22 14:47:39 UTC │
    ├── DistributionTask        │ 1459 │ completed │             │ Bva         │ 2020-05-22 14:47:39 UTC │
    ├── JudgeAssignTask         │ 1460 │ completed │             │ BVAAABSHIRE │ 2020-06-01 14:34:38 UTC │
    ├── JudgeDecisionReviewTask │ 2493 │ cancelled │ BVAAABSHIRE │ BVAAABSHIRE │ 2020-06-01 14:52:39 UTC │
    │   └── AttorneyTask        │ 2494 │ cancelled │ BVAAABSHIRE │ BVAEERDMAN  │ 2020-06-01 14:52:39 UTC │
    ├── JudgeAssignTask         │ 2495 │ cancelled │             │ BVAAABSHIRE │ 2020-06-01 14:53:15 UTC │
    ├── JudgeAssignTask         │ 2496 │ completed │ BVAAABSHIRE │ BVAAABSHIRE │ 2020-06-01 18:17:36 UTC │
    └── JudgeDecisionReviewTask │ 2523 │ on_hold   │ BVARDUNKLE  │ BVAAABSHIRE │ 2020-06-01 18:17:36 UTC │
        └── AttorneyTask        │ 2524 │ assigned  │ BVARDUNKLE  │ BVAEERDMAN  │ 2020-06-01 18:17:36 UTC │

=> nil
task.cancelled!
```
1. Ensure this does not close the judge review task and does not open a judge assign task
```ruby
task.appeal.reload.treee
Appeal 545 (direct_review)       ID   STATUS    ASGN_BY     ASGN_TO     UPDATED_AT
└── RootTask                     1458 on_hold               Bva         2020-05-22 14:47:39 UTC
    ├── DistributionTask         1459 completed             Bva         2020-05-22 14:47:39 UTC
    ├── JudgeAssignTask          1460 completed             BVAAABSHIRE 2020-06-01 14:34:38 UTC
    ├── JudgeDecisionReviewTask  2493 cancelled BVAAABSHIRE BVAAABSHIRE 2020-06-01 14:52:39 UTC
    │   └── AttorneyTask         2494 cancelled BVAAABSHIRE BVAEERDMAN  2020-06-01 14:52:39 UTC
    ├── JudgeAssignTask          2495 cancelled             BVAAABSHIRE 2020-06-01 14:53:15 UTC
    ├── JudgeAssignTask          2496 completed BVAAABSHIRE BVAAABSHIRE 2020-06-01 18:17:36 UTC
    └── JudgeDecisionReviewTask  2523 assigned  BVARDUNKLE  BVAAABSHIRE 2020-06-02 19:45:16 UTC
        └── AttorneyTask         2524 cancelled BVARDUNKLE  BVAEERDMAN  2020-06-02 19:45:16 UTC
```

### User Facing Changes
